snare/tests/test_cloner_get_body.py

```diff
@@ -48,113 +51,228 @@ def setUp(self):
+
+        if not self.handler.runner:
+            raise Exception("Error initializing Cloner!")
+        self.filename, self.hashname = self.handler.runner._make_filename(yarl.URL(self.root))
         self.expected_content = '<html><body><a href="/test"></a></body></html>'
 
         self.meta = {
-            "/index.html": {
-                "hash": "d1546d731a9f30cc80127d57142a482b",
-                "headers": [{"Content-Type": "text/html"}],
+            "/": {
+                "hash": "6666cd76f96956469e7be39d750cc7d9",
+                "headers": [{"Server": "Apache"}],
+                "content_type": "text/html",
             },
             "/test": {
                 "hash": "4539330648b80f94ef3bf911f6d77ac9",
-                "headers": [{"Content-Type": "text/html"}],
+                "headers": [{"Server": "Apache"}],
+                "content_type": "text/html",
             },
         }
 
         async def test():
-            await self.handler.new_urls.put((yarl.URL(self.root), 0))
-            await self.handler.get_body(self.session)
+            if not self.handler.runner:
+                raise Exception("Error initializing Cloner!")
+            await self.handler.runner.new_urls.put({"url": yarl.URL(self.root), "level": 0, "try_count": 0})
+            await self.handler.runner.get_body(self.session)
 
         with self.assertLogs(level="DEBUG") as log:
             self.loop.run_until_complete(test())
-            self.assertIn("DEBUG:snare.cloner:Cloned file: /test", "".join(log.output))
+            self.assertIn("DEBUG:snare.cloner:Cloned file: /test", "\n".join(log.output))
 
         with open(os.path.join(self.target_path, self.hashname)) as f:
             self.return_content = f.read()
 
         self.assertEqual(self.return_content, self.expected_content)
         self.assertEqual(
-            self.handler.visited_urls[-2:],
+            self.handler.runner.visited_urls[-2:],
             ["http://example.com/", "http://example.com/test"],
         )
-        self.assertEqual(self.handler.meta, self.meta)
+        self.assertEqual(self.handler.runner.meta, self.meta)
+
+        for p in patches:
+            p.stop()
 
     def test_get_body_css_validate(self):
-        aiohttp.ClientResponse._headers = {"Content-Type": "text/css"}
 
         self.css_validate = True
-        self.handler = Cloner(self.root, self.max_depth, self.css_validate)
+        self.handler = CloneRunner(self.root, self.max_depth, self.css_validate)
         self.content = b""".banner { background: url("/example.png") }"""
-        aiohttp.ClientResponse.read = AsyncMock(return_value=self.content)
+
+        patches = []
+        patches.append(patch("aiohttp.ClientResponse._headers", new={"Content-Type": "text/css", "Server": "Apache"}))
+        patches.append(patch("aiohttp.ClientResponse.read", new=AsyncMock(return_value=self.content)))
+        for p in patches:
+            p.start()
+
+        self.session.get = AsyncMock(
```

Comment:
you have similar mock before, can it be reused?

Reply:
I think each of the mocks are unique, so they can't be reduced, but to reduce the visual clutter, I have introduced a separate `create_client_response` method which takes the URL and event loop as arguments to return the ClientResponse object.